src/mapContainer.ts

```diff
@@ -695,6 +931,67 @@ export class MapContainer {
                     ev.stopPropagation();
                 });
         });
+        newMarker.on('move', (event: leaflet.LeafletEvent) => {
+            let oldMarkerLocation = marker.location.toString();
+            let vertices = this.display.vertices;
+            if (vertices.has(oldMarkerLocation)) {
+                let vertexToUpdate = vertices.get(oldMarkerLocation);
+                let newLatLng = newMarker.getLatLng().clone();
+                for (let edge of vertexToUpdate) {
+                    if (edge.polyline) {
+                        if (edge.v1.toString() === oldMarkerLocation) {
+                            edge.v1 = newLatLng;
+                        } else if (edge.v2.toString() === oldMarkerLocation) {
+                            edge.v2 = newLatLng;
+                        }
+                        edge.polyline.remove();
+                        this.display.polylines = this.display.polylines.filter(
+                            (x) => x !== edge.polyline
+                        );
+                        let polyline = leaflet.polyline([edge.v1, edge.v2], {
+                            color: this.settings.edgeColor,
+                            weight: 1,
+                        });
+                        edge.polyline = polyline;
+                        polyline.addTo(this.display.map);
+                        this.display.polylines.push(polyline);
+                    }
+                }
+                marker.location = newLatLng;
+                vertices.delete(oldMarkerLocation);
+                vertices.set(marker.location.toString(), vertexToUpdate);
+            }
+        });
+        newMarker.on('moveend', async (event: leaflet.LeafletEvent) => {
+            const content = await this.app.vault.read(marker.file);
+            let newLat = marker.location.lat;
+            let newLng = marker.location.lng;
+            if (marker.isFrontmatterMarker) {
+                let parsed = frontMatter<FrontMatterAttributes>(content);
+                parsed.attributes.location = [newLat, newLng];
+                let updatedContent = `---\n${yaml.dump(
```

Comment:
Does this work when the YAML has other attributes too? If I understand correctly, it rewrites the entire thing

Reply:
Yes, this works when the YAML has other attributes. After frontMatter parses the content, all of the front matter attributes end up in the attributes object, so parsed.attributes.location = X only updates the location property. Since the entire attributes object is then passed to yaml.dump, any pre-existing attributes are preserved.

![image](https://github.com/esm7/obsidian-map-view/assets/153950304/b596fd9b-ed39-4414-aba1-231bb7daee4c)

![image](https://github.com/esm7/obsidian-map-view/assets/153950304/50d5d8ae-55b3-4778-9006-6c9642563951)